src/components/booking/ExpandedSessionCard.tsx

```diff
@@ -1,55 +1,54 @@
-/**
- * @author Angela Guo <aguo921@aucklanduni.ac.nz>
- */
-
 import { useState } from "react";
 import { IoCheckmarkCircle } from "react-icons/io5";
 
 import { Card } from "../Card";
 import { LevelSelector } from "./LevelSelector";
 
 interface ExpandedSessionCardProps {
-  id: string;
+  day: weekday;
   startTime: Date;
   endTime: Date;
   location: string;
   address: string;
-  level?: string;
-  defaultLevel?: string;
-  setLevel: (level: string | undefined) => void;
 }
-
-const weekday = [
-  "Sunday",
-  "Monday",
-  "Tuesday",
-  "Wednesday",
-  "Thursday",
-  "Friday",
-  "Saturday",
-];
+enum weekday {
+  Monday = "Monday",
+  Tuesday = "Tuesday",
+  Wednesday = "Wednesday",
+  Thursday = "Thursday",
+  Friday = "Friday",
+  Saturday = "Saturday",
+  Sunday = "Sunday",
+}
 
 export const ExpandedSessionCard = ({
+  day,
   startTime,
   endTime,
   location,
   address,
-  level,
-  setLevel,
-  defaultLevel,
-}: Omit<ExpandedSessionCardProps, "id">) => {
-  const [isOpen, setIsOpen] = useState(false);
-  const [pendingLevel, setPendingLevel] = useState<string | undefined>(level);
+}: ExpandedSessionCardProps) => {
+  const [showModal, setShowModal] = useState(false);
+  const [pendingLevel, setPendingLevel] = useState<string | undefined>(
+    undefined,
+  );
+
+  const handleButtonClick = () => {
+    setShowModal(true);
+  };
 
-  const dayOfWeek = weekday[startTime.getDay()];
+  const handleCloseModal = () => {
+    setShowModal(false);
+    setPendingLevel(pendingLevel);
+  };
 
   return (
     <Card className="relative font-normal">
       <div className="rounded-t-md bg-blue-600 px-6 py-4 pr-10 drop-shadow-lg">
```

Comment:
```suggestion
      <div className="rounded-t-md bg-primary px-6 py-4 pr-10 drop-shadow-lg">
```
use the variable colours instead. See SelectSessionCard for examples. You can find the variables in globals.css. Refer to figma for what colours to use.